coredocument/coredocument.go

```diff
@@ -150,33 +151,52 @@ func GetDocumentSigningTree(document *coredocumentpb.CoreDocument) (tree *proofs
 
 // PrepareNewVersion creates a copy of the passed coreDocument with the version fields updated
 // Adds collaborators and fills salts
-// Note: ignores any collaborators in the oldCD
+// Note: new collaborators are added to the list with old collaborators.
 func PrepareNewVersion(oldCD coredocumentpb.CoreDocument, collaborators []string) (*coredocumentpb.CoreDocument, error) {
-	newCD, err := NewWithCollaborators(collaborators)
+	ncd := New()
+	ucs, err := fetchUniqueCollaborators(oldCD.Collaborators, collaborators)
+	if err != nil {
+		return nil, errors.New("failed to decode collaborator: %v", err)
+	}
+
+	cs := oldCD.Collaborators
+	for _, c := range ucs {
+		c := c
```

Comment:
beautiful syntax 😉

Reply:
This is to avoid the slice reference of the previous element